### ISSUE TITLE
The three `declaration-colon-*` rules take a colon the property itself spells for the declaration's own, and one of them writes a line break into a comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#421]
### What minimal example or steps are needed to reproduce the bug?

All three `declaration-colon-*` rules look for the declaration's colon in the text standing in front of the value — the property and `raws.between` together — and take the first colon they find there. A property may spell a colon of its own, an escaped one, and that colon comes first:

```css
a { b\:c: pink; }
```

The escape is ordinary CSS: an identifier may hold one, and `lightningcss` reads that declaration and prints it back unchanged. But the rules read the `\:` of the property as the declaration's colon, and everything they do afterwards is measured from a character that is not the one they are about.

`declaration-colon-space-before` with `always` writes a space in front of the property's colon rather than the declaration's, and the run it wrote is not the run it reads on the next pass, so the file grows by a space on every run:

```css
a { b\:c : pink; }
a { b\:c  : pink; }
a { b\:c   : pink; }
```

`declaration-colon-space-after` with `always` reports the declaration above, whose single space is already in place, and its fix writes nothing at all — the position it counts from the property's colon falls in front of `raws.between`, where there is nothing to slice. So `--fix` runs clean while the warning stands, run after run.

`declaration-colon-newline-after` with `always` writes its break in front of the property's colon:

```css
a { b\:c

:/*c*/; }
```

and where the value opens on a comment, the break lands inside that comment's text and the file stops parsing:

```css
a { b\:c:/*c*/pink; }
```

comes back as

```css
a { b\:c:/*c*
/pink; }
```

and the next run reports `Unclosed comment`.

A sweep of 2625 forms — five spellings of a property holding an escaped colon, five widths of run behind the colon, five tails of the value, three syntaxes and the seven configurations the three rules accept — has 57 forms whose `--fix` output does not parse, 939 on which `--fix` never settles, and 1215 that carry a warning `--fix` has already claimed to answer. Every one of the 57 is `declaration-colon-newline-after` with `always` over a value opening on a comment.

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": [
		"@stylistic/stylelint-plugin"
	],
	"rules": {
		"@stylistic/declaration-colon-space-before": "always"
	}
}
```

### What did you expect to happen?

A colon the property spells opens no declaration, so it is neither reported nor written near, and whatever the declaration's own colon earns settles on the first run of `--fix`.

### What actually happened?

The property's colon is taken for the declaration's: one rule grows the file by a space on every run, another reports a declaration it then leaves exactly as it was, and the third writes a line break into the text of a comment and leaves a file no parser will read.

### Notes

The declaration's own colon is never in the property. PostCSS files everything the file spells between the property and the value in `raws.between`, the colon among it, so the search belongs over that raw alone — and a comment inside it may still hold a colon of its own, which is why the search stays the comment-aware one [#92](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/92) put there.

Found while closing [#371](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/371), whose branch answers it: that branch widens the text the rules read to the end of the value, which would carry the same misreading into every value holding no word of its own, and bounding the search to `raws.between` is what takes both away.
